contents/staticAnalysis/PMD.md

```diff
@@ -0,0 +1,73 @@
+## PMD
+PMD is a static analyser for Java, JavaScript, Salesforce.com Apex, PL/SQL, Apache Velocity, XML, XSL. The Copy-pate-detector ([CPD Find duplicated code](http://pmd.sourceforge.net/pmd-4.3.0/cpd.html)) is also included as an add-on in PMD.
+
+### Features
+PMD supports checking rules for the following languages.
+
+- [Java](https://pmd.github.io/pmd-5.5.3/pmd-java/index.html) (includes [JSP](https://pmd.github.io/pmd-5.5.3/pmd-java/index.html) with [limitations](https://pmd.github.io/pmd-5.5.3/pmd-jsp/index.html))
+- [JavaScript](https://pmd.github.io/pmd-5.5.3/pmd-javascript/index.html)
+- [Apex](https://pmd.github.io/pmd-5.5.3/pmd-apex/index.html)
+- [PL/SQL](https://pmd.github.io/pmd-5.5.3/pmd-plsql/index.html)
+- [Velocity](https://pmd.github.io/pmd-5.4.1/pmd-vm/index.html)
+- [XML and XSL](https://pmd.github.io/pmd-5.5.3/pmd-xml/index.html)
+
+PMD doesn't support checking rules sets for the following languages. Only Copy/Paste Detector ([CPD](http://pmd.sourceforge.net/pmd-4.3.0/cpd.html)) is supported for them.
+
+- [C++](https://pmd.github.io/pmd-5.5.3/pmd-cs/index.html)
+- [C#](https://pmd.github.io/pmd-5.5.3/pmd-cpp/index.html)
+- [Fortran](https://pmd.github.io/pmd-5.5.3/pmd-fortran/index.html)
+- [Go](https://pmd.github.io/pmd-5.5.3/pmd-go/index.html)
+- [Groovy](https://pmd.github.io/pmd-5.5.3/pmd-groovy/index.html)
+- [Matlab](https://pmd.github.io/pmd-5.5.3/pmd-matlab/index.html)
+- [Objective-C](https://pmd.github.io/pmd-5.5.3/pmd-objectivec/index.html)
+- [Perl](https://pmd.github.io/pmd-5.5.3/pmd-perl/index.html) (only very basic CPD support)
+- [PHP](https://pmd.github.io/pmd-5.5.3/pmd-php/index.html)
+- [Python](https://pmd.github.io/pmd-5.5.3/pmd-python/index.html)
+- [Ruby](https://pmd.github.io/pmd-5.5.3/pmd-ruby/index.html)
+- [Swift](https://pmd.github.io/pmd-5.5.3/pmd-swift/index.html)
+- [Scala](https://pmd.github.io/pmd-5.5.3/pmd-scala/index.html)
+
+### Limitation
+Limitations are almost the same as [CheckStyle](checkStyle.md).
+
+- The examined code have to be compilable. The reason is described in [How does it work](#how-does-it-work) section.
+- Files will be examined one by one, there is no check to check multiple files at the same time (e.g. you cannot determine the full inheritance hierarchy of a class)
```

Comment:
Can consider rewording it along the lines of "Files will be examined one by one. This might be an issue as you cannot determine the full inheritance hierarchy of a class by checking multiple files at the same time."

Or minimally rephrase "no check to check"